Задание спринта 22 (добавлены сервисы payment, delivery, order)

PUprojects opened the pull request:
Возможно, где-то не совсем понял логику. Почти ничего не проверял. Конфигурация новых сервисов читается, все сервисы регистриуются в eureka.

PUprojects committed:
Задание спринта 22 (добавлены сервисы payment, delivery, order)
Возможно, где-то не совсем понял логику. Почти ничего не проверял.
Конфигурация новых сервисов читается, все сервисы регистриуются в eureka.

PUprojects committed:
Забыл добавить методы к фулбэку warehouse

bismuthD commented on commerce/order/pom.xml:
```diff
@@ -0,0 +1,113 @@
+
+    <artifactId>order</artifactId>
+
+    <properties>
```
Свойства с версией и кодировкой лучше вынести в самый корневой pom, т.к. они будут одинаковые у всех модулей

bismuthD commented on commerce/interaction-api/src/main/java/ru/yandex/practicum/commerce/contract/delivery/DeliveryOperations.java:
```diff
@@ -0,0 +1,29 @@
+@Validated
+public interface DeliveryOperations {
+    @PutMapping("/api/v1/delivery")
+    DeliveryDto addDelivery(@RequestBody DeliveryDto newDelivery);
```
стоит добавить @Valid

bismuthD commented on commerce/interaction-api/src/main/java/ru/yandex/practicum/commerce/contract/delivery/DeliveryOperations.java:
```diff
@@ -0,0 +1,29 @@
+    void filedDelivery(@RequestBody UUID deliveryId);
+
+    @PostMapping("/api/v1/delivery/cost")
+    BigDecimal getDeliveryCoast(@RequestBody OrderDto order);
```
стоит добавить @Valid
Для остальных интерфейсов-клиентов тоже добавь валидацию

bismuthD commented on commerce/payment/src/main/java/ru/yandex/practicum/commerce/payment/service/PaymentServiceImpl.java:
```diff
@@ -0,0 +1,85 @@
+
+    @Override
+    public BigDecimal getTotalCoast(OrderDto order) {
+        final double TAX_RATE = 1.1;
```
Здесь также все расчеты стоимости лучше делать с BigDecimal

bismuthD reviewed (commented):
Отличная работа. Архитектура, реализация модулей payment, delivery и order реализованы корректно. Оставил несколько комментариев для правок

PUprojects committed:
Исправлены замечания ревьюэра:
- секция properties вынесена в корневой pom файл проекта;
- добавлена валидация в интерфейсах;
- в сервисе payment расчёты переведены на BigDecimal

PUprojects committed:
Убрано ещё одно упоминание Java7. В головном pom файле проекта добавлена версия Java 22.

PUprojects committed:
Вернул версию Java 21, т.к. 22 не поддерживается Avro
